src/darwin/Framework/CHIP/templates/MTRBaseClusters-src.zapt

```diff
@@ -25,15 +25,17 @@ using chip::SessionHandle;
 {{#chip_client_clusters includeAll=true}}
 @implementation MTRBaseCluster{{asUpperCamelCase name}}
 
-- (instancetype)initWithDevice:(MTRBaseDevice *)device endpoint:(uint16_t)endpoint queue:(dispatch_queue_t)queue
+- (instancetype)initWithDevice:(MTRBaseDevice *)device endpoint:(NSNumber *)endpoint queue:(dispatch_queue_t)queue
 {
     if (self = [super initWithQueue:queue]) {
         if (device == nil) {
             return nil;
         }
 
         _device = device;
-        _endpoint = endpoint;
+        {{!TODO consider range-checking the incoming number to make sure it's
+                actually in the uint16_t range}}
```

Comment:
Not for this patch, but if we start doing more and more `NSNumber` conversion to c-types, I wonder if we should end up doing some helpers that does both the conversion and the range check.

Reply:
Yes, that might be nice once we start doing it.